Develop

MatveyStrakhov opened the pull request.

MatveyStrakhov committed:
Initial commit: branch ready

MatveyStrakhov committed:
Initial commit: branch ready checkstyle passed

MatveyStrakhov committed:
feat: added director to database

MatveyStrakhov committed:
feat: added Director in model

MatveyStrakhov committed:
feat: added Director in model.Film, added DirectorMapper

EryOller committed:
Функциональность "Отзывы" - Добавить сущность Review

EryOller committed:
Функциональность "Отзывы" - Реализовать ReviewMapper

MatveyStrakhov committed:
add-director: feat: added director parameter to all film functions.

MatveyStrakhov committed:
add-director: feat: added DirectorService

MatveyStrakhov committed:
add-director: feat: added DirectorController, added deleteDirector function

EryOller committed:
Функциональность "Отзывы" - Реализовать ReviewDao

EryOller committed:
Функциональность "Отзывы" - Реализовать таблицы reviews, films_reviews, likes_reviews, users_reviews в БД

MatveyStrakhov committed:
add-director: feat: added getFilmsByDirector to Films chain

EryOller committed:
Функциональность "Отзывы" - Реализовать ReviewService

MatveyStrakhov committed:
add-director: feat: added DirectorControllerTest, rework of all tests

MatveyStrakhov committed:
add-director: fix: fixed bugs with director-related tests, fixed FilmsExtractor

MatveyStrakhov committed:
add-director: style: checkstyle passed

MatveyStrakhov committed:
add-director: style: checkstyle second attempt

MatveyStrakhov committed:
add-director: style: checkstyle third attempt

MatveyStrakhov committed:
add-director: style: checkstyle 4th attempt: removed unused import

EryOller committed:
Функциональность "Отзывы" - Реализовать ReviewController

MatveyStrakhov committed:
add-director: fix: Director name bug, added @NotBlank

MatveyStrakhov committed:
add-director: fix: FilmsByDirectorByYear is now ASC

MatveyStrakhov committed:
Merge pull request #9 from MatveyStrakhov/add-director

Add director

EryOller committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-reviews

# Conflicts:
#	src/main/resources/schema.sql

PVVik committed:
feat: "добавила функционал поиска общих фильмов"

MatveyStrakhov committed:
add-director: fix: minor bug fixed in getPopularFilms

Taratorin committed:
добавлен функционал поиска фильмов по названию и режиссеру

MatveyStrakhov committed:
Merge pull request #10 from MatveyStrakhov/add-director

add-director: fix: minor bug fixed in getPopularFilms

Taratorin committed:
Merge pull request #12 from MatveyStrakhov/add-search

добавлен функционал поиска фильмов по названию и режиссеру

EryOller committed:
Функциональность "Отзывы" - Отладить код

EryOller committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-reviews

EryOller committed:
Функциональность "Отзывы" - Отладить код

EryOller committed:
Функциональность "Отзывы" - Привести код к стилю

EryOller committed:
Функциональность "Отзывы" - Привести код к соответствующему стилю

EryOller committed:
Функциональность "Отзывы" - Привести код к соответствующему стилю

EryOller committed:
Функциональность "Отзывы" - Привести код к соответствующему стилю

EryOller committed:
Merge pull request #15 from MatveyStrakhov/add-reviews

Add reviews

Taratorin committed:
добавить функционал ленты собитий

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-feed

# Conflicts:
#	src/main/resources/schema.sql

Taratorin committed:
добавить функционал ленты событий для отзывов пользователей

YaXynA committed:
GetMapping getPopularFilms заменен на перегрузку findPopularFilms добавлен DeletMapping

YaXynA committed:
метод getPopularFilms заменен на перегрузку findPopularFilms добавлен метод удаления

YaXynA committed:
метод getPopularFilms заменен на перегрузку findPopularFilms добавлен метод удаления

YaXynA committed:
Переделан DeleteMap

YaXynA committed:
Переделан метод DeleteUser

Taratorin committed:
bugs fixed

YaXynA committed:
Убрал 2 теста которые мне мешали

YaXynA committed:
checkStyle

YaXynA committed:
checkStyle

YaXynA committed:
checkStyle

YaXynA committed:
Убрал тест который мешает

YaXynA committed:
Убрал тест который мешает

Taratorin committed:
Merge pull request #16 from MatveyStrakhov/add-feed

Add feed

YaXynA committed:
Merge branch 'develop' of github.com:MatveyStrakhov/java-filmorate into add-most-populars

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/impl/UserDbStorage.java

YaXynA committed:
Удаление из feeds

YaXynA committed:
Merge pull request #17 from MatveyStrakhov/add-most-populars

Add most populars

PVVik committed:
feat: "добавила функционал поиска общих фильмов"

EryOller committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-common-films

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmStorage.java

EryOller committed:
Функциональность "Общие фильмы" - Исправить sql запрос по получению фильмов

MatveyStrakhov committed:
add-recommendations: feat: added recommendations

MatveyStrakhov committed:
add-recommendations: style: minor fixes in code

EryOller committed:
Merge pull request #18 from MatveyStrakhov/add-common-films

Add common films

MatveyStrakhov committed:
Merge remote-tracking branch 'origin/develop' into add-recommendations

MatveyStrakhov committed:
Merge pull request #19 from MatveyStrakhov/add-recommendations

Add recommendations

Taratorin committed:
bugs fixed

Taratorin committed:
Merge pull request #20 from MatveyStrakhov/add-feed

bugs fixed

Taratorin committed:
удалить неиспользуемый метод event2;

Taratorin committed:
удалить неиспользуемые импорты;

Taratorin committed:
Merge pull request #22 from MatveyStrakhov/add-feed

удалить неиспользуемый метод event2;

MatveyStrakhov committed:
Delete src/main/java/ru/yandex/practicum/filmorate/storage/FilmDbStorage.java

deleting empty file

MatveyStrakhov committed:
Update README.md

Adding new diagram

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -0,0 +1,301 @@
+                "LEFT JOIN directors AS d ON fd.director_id=d.director_id " +
+                "WHERE d.director_id=";
+        String sqlQuery;
+        if (sortBy.equals("likes")) {
```
Хорошо бы заменить на switch и enum

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -0,0 +1,301 @@
+                "LEFT JOIN genre AS g ON fg.genre_id=g.genre_id\n" +
+                "LEFT JOIN film_director AS fd ON f.id=fd.film_id\n" +
+                "LEFT JOIN directors AS d ON fd.director_id=d.director_id\n";
+        String sqlEnd = "GROUP BY F.ID\n" + "ORDER BY COUNT DESC;";
```
Лучше конкатенацию для всего запроса строить с помощью StringBuilder

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -0,0 +1,301 @@
+                "WHERE LOWER(d.DIRECTOR) LIKE '%" + queryLower + "%'\n" +
+                "OR LOWER(f.NAME) LIKE '%" + queryLower + "%'\n" + sqlEnd;
+        switch (by) {
+            case "title":
```
Можно заменить на enum

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -0,0 +1,301 @@
+                "LEFT JOIN film_director AS fd ON f.ID = fd.film_id " +
+                "LEFT JOIN directors AS d ON fd.director_id = d.director_id " +
+                "WHERE f.ID IN (SELECT DISTINCT FILM_ID FROM " +
+                " (SELECT fg2.FILM_ID , COUNT(l2.USER_ID) AS c " +
```
Внутренний select лучше заменить на with или выполнить отдельным запросом

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/RecommendationsDao.java:
```diff
@@ -0,0 +1,158 @@
+        Set<Film> films = filmStorage.returnAllFilms().stream().sorted(Comparator.comparing(Film::getId, Integer::compareTo)).collect(Collectors.toCollection(LinkedHashSet::new));
+        Set<Integer> users = userStorage.returnAllUsers().stream().map(User::getId).collect(Collectors.toSet());
+        List<Like> likes = getLikes();
+        for (Like like : likes) {
```
Можно заменить на forEach

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/RecommendationsDao.java:
```diff
@@ -0,0 +1,158 @@
+    private final LikesMapper likesMapper;
+    private final JdbcTemplate jdbcTemplate;
+
+    private List<Like> getLikes() {
```
Приватные методы обычно пишутся в конце публичных

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/RecommendationsDao.java:
```diff
@@ -0,0 +1,158 @@
+            return getRecommendedFilms(inputData, predictionMatrix, userId);
+        }
+
+        private static void buildDifferencesMatrix(Map<Integer, HashMap<Film, Double>> data) {
```
Пожелание по методам
1) Слишком много вложенных for, по возможности лучше зарефакторить
2) Переменные лучше называть понятно, не e, j, i

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/RecommendationsDao.java:
```diff
@@ -0,0 +1,158 @@
+
+    private static class SlopeOne {
+
+        private static Map<Film, Map<Film, Double>> diff = new HashMap<>();
```
Лучше эти переменные сделать не статическими

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/EventDao.java:
```diff
@@ -0,0 +1,35 @@
+@Component
+public class EventDao {
+
+    private static JdbcTemplate jdbcTemplate;
```
В бинах лучше не использовать статические поля других бинов

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/EventDao.java:
```diff
@@ -0,0 +1,35 @@
+        EventDao.jdbcTemplate = jdbcTemplate;
+    }
+
+    public static void eventAdd(int entityId, String eventType, String operation, int userId) {
```
У бинов методы должны быть не статическими

EryOller committed:
Функциональность "Отзывы" - Добавить интерфейсы для сервиса и dao. Переместить логику в сервисы. Добавить логирование в коньтроллеры.

Taratorin committed:
в классе EventDao удалить статические поля и методы;
в классах FilmDbStorage и UserDbStorage добавить аннотацию @AllArgsConstructor и поле eventDao;
в классе ReviewDao добавить поле eventDao;

Taratorin committed:
конкатенацию для всего запроса строить с помощью StringBuilder;

Taratorin committed:
логику валидации в методе getFilmsByDirector вынести в сервис filmService;

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-director

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

Taratorin committed:
bug fixed;

Taratorin committed:
логику валидации вынести в filmService;

Taratorin committed:
не возвращать объект в методе deleteUser класса UserController;

EryOller committed:
Merge pull request #26 from MatveyStrakhov/add-reviews

Функциональность "Отзывы" - Добавить интерфейсы для сервиса и dao. Пе…

Taratorin committed:
создать интерфейс для методов класса DirectorService;

Taratorin committed:
исправить стиль SQL в классе UserDbStorage;

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-feed

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/storage/impl/ReviewDao.java

Taratorin committed:
добавить поле eventDao в класс ReviewDao;

Taratorin committed:
перенести вызов методов upUseful и downUseful в класс ReviewDao;

Taratorin committed:
не возвращать объект в методе deleteUser класса UserService;

Taratorin committed:
использовать switch в методе getFilmsByDirector класса FilmDbStorage;

Taratorin committed:
использовать forEach в методе getRecommendedFilms класса RecommendationsDao;

Taratorin committed:
переместить приватный метод getLikes после публичных;

Taratorin committed:
убрать статические переменные внутреннего класса SlopeOne в классе RecommendationsDao;

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-director

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java

Taratorin committed:
убрать статические переменные внутреннего класса SlopeOne в классе RecommendationsDao;

Taratorin committed:
убрать неиспользуемые импорты;

Taratorin committed:
bug fixed;

PVVik committed:
fix: "исправила замечание по пр"

PVVik committed:
fix: "исправила замечание по пр"

PVVik committed:
Merge pull request #33 from MatveyStrakhov/add-most-populars

Add most populars

PVVik committed:
fix: "исправила замечание по пр"

EryOller committed:
Merge pull request #30 from MatveyStrakhov/add-feed

Add feed

PVVik committed:
Merge pull request #34 from MatveyStrakhov/add-remove-endpoint

fix: "исправила замечание по пр"

EryOller committed:
fix: Добавил аннотация для конструктора

EryOller committed:
Merge pull request #35 from MatveyStrakhov/add-reviews

fix: Добавил аннотация для конструктора

EryOller committed:
fix: Добавил интерфейсы для DAO

EryOller committed:
Merge pull request #36 from MatveyStrakhov/add-reviews

fix: Добавил интерфейсы для DAO

PVVik committed:
feat: "добавила интерфейсы к сервисам"

PVVik committed:
feat: "добавила интерфейсы к сервисам"

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-director

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

Taratorin committed:
удалить неиспользуемые импорты;

Taratorin committed:
Merge pull request #39 from MatveyStrakhov/add-search

Add search

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-director

EryOller committed:
Merge pull request #38 from MatveyStrakhov/add-recommendations

feat: "добавила интерфейсы к сервисам"

Taratorin committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-director

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/ReviewController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/DirectorService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/impl/RecommendationsDao.java

Taratorin committed:
исправить конфликт;

Taratorin committed:
Merge pull request #31 from MatveyStrakhov/add-director

Add director

Taratorin committed:
переместить интерфейс;

EryOller committed:
fix: Исправление стиля sql запросов

Taratorin committed:
положительный count валидировать с помощью аннотации на уровне контроллера;

Taratorin committed:
убрать избыточные проверки в FilmService;

Taratorin committed:
удалить неиспользуемые импорты;

EryOller committed:
Merge pull request #41 from MatveyStrakhov/add-director

Add director

EryOller committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-reviews

EryOller committed:
fix: Добавил логирование в контроллеры. Перенес логику из контроллеров по работе с пользователями и фильмами. Добавил в README список выполненных задач.

EryOller committed:
fix: Удаление не используемого коммита

EryOller committed:
Merge pull request #42 from MatveyStrakhov/add-reviews

Add reviews

MatveyStrakhov committed:
add-recommendations: testing new logic

MatveyStrakhov committed:
add-recommendations:refactor: refactored RecommendationsDao, FilmsExtractor

MatveyStrakhov committed:
Merge branch 'develop' of https://github.com/MatveyStrakhov/java-filmorate into add-recommendations

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/storage/impl/RecommendationsDao.java

MatveyStrakhov committed:
add-recommendations:refactor: updated

MatveyStrakhov committed:
add-recommendations:refactor: refactored tests

EryOller committed:
Merge pull request #43 from MatveyStrakhov/add-recommendations

Add recommendations
